opencog/nlp/chatbot/README

```diff
@@ -82,9 +37,13 @@ running ./opencog-server.sh from the main RelEx directory. (You must
 compile RelEx first, by installing java and ant, and saying "ant" to
 build RelEx).
 
-The parse server listens on port 4444: it will read plain-text input,
-and generates parsed text in opencog format.  The chatbot needs this
-server to process text.
+The parse server listens on port 4444: it will read plain-text input and
+generates parsed text in OpenCog format. It uses RelEx2Logic rules 
```

Comment:
Ah, just to clarify there . . . you mean, I ported all the rules, which were written in their own syntax in a text rule file;
the java code you're talking about was what applied the the rules, and did a couple of other functions that also need porting,
such as creating the InterpretationNodes.  As for what I did port, be aware that there were a handful of old rules that
I didn't port or becausethey required the output of that java code to function.  I made extensive comments about those
exceptions in the rule fileitself, which revised version should be available now in the repostitory, and should be examined
for those comments by whoever ports the java and / or refactors the rule.  I will also discuss some of that in the coming 
part II of the R2L documentation that Amen just posted for me.

--------------------------------------------
On Mon, 7/13/15, bgoertzel <notifications@github.com> wrote:

 Subject: Re: [opencog] Dialog (#1675)
 To: "opencog/opencog" <opencog@noreply.github.com>
 Date: Monday, July 13, 2015, 4:44 AM
 
 In opencog/nlp/chatbot/README:
 > @@ -82,9 +37,13 @@ running
 ./opencog-server.sh from the main RelEx directory. (You must
 >  compile RelEx first, by installing java and ant, and
 saying "ant" to
 >  build RelEx).
 >  
 > -The parse server listens on port 4444: it will read
 plain-text input,
 > -and generates parsed text in opencog format.  The
 chatbot needs this
 > -server to process text.
 > +The parse server listens on port 4444: it will read
 plain-text input and
 > +generates parsed text in OpenCog format. It uses
 RelEx2Logic rules 
 
 Linas,
 
 I believe that in the currently working system, R2L is still
 split between
 some code in the RelEx output generator, and some Scheme
 functions.
 
 All the stuff in the Java R2L code has been ported to Scheme
 functions now
 (by Aaron)....  However, nobody has yet written a process to
 actually apply
 these functions.  Mandeep is going to do this during the
 coming week.  But
 for the moment, I believe Rodas is right that the NLP
 pipeline is using the
 old partly-Java R2L pipeline.  Unless you changed something
 and I didn't
 notice ;)
 …
 --ben
 On Mon, Jul 13,
 2015 at 1:27 AM, Linas Vepštas
 <notifications@github.com> wrote:
  In opencog/nlp/chatbot/README
  <https://github.com/opencog/opencog/pull/1675#discussion_r34434183>:
 
  > @@ -82,9 +37,13 @@ running ./opencog-server.sh from
 the main RelEx directory. (You must
  >  compile RelEx first, by installing java and ant, and
 saying "ant" to
  >  build RelEx).
  >
  > -The parse server listens on port 4444: it will read
 plain-text input,
  > -and generates parsed text in opencog format.  The
 chatbot needs this
  > -server to process text.
  > +The parse server listens on port 4444: it will read
 plain-text input and
  > +generates parsed text in OpenCog format. It uses
 RelEx2Logic rules
 
  This is not quite correct; R2L iis now done in opencog, and
 not in relex.
 
  —
  Reply to this email directly or view it on GitHub
  <https://github.com/opencog/opencog/pull/1675/files#r34434183>.
 
 
 -- 
 Ben Goertzel, PhD
 http://goertzel.org
 
 "The reasonable man adapts himself to the world: the
 unreasonable one
 persists in trying to adapt the world to himself. Therefore
 all progress
 depends on the unreasonable man." -- George Bernard
 Shaw
 
 
 —
 Reply to this email directly or view
 it on GitHub.